README.md

```diff
@@ -115,3 +115,102 @@ that sysctl settings which are recommended for the currently running kernel
 are applied by **systemd-sysctl.service** at boot time. These settings are
 shipped in the file `/boot/sysctl.conf-$(uname -r)`, which is part of the
 kernel package.
+
+
+# BLS and ESP
+
+There scripts are generating boot entries (via perl-Bootloader), new
+initrds (via dracut) and updating the kernel module dependency lists
+(via depmod).  If we are in a system using the boot entries defined in
+the bootloader specification (BLS), then we need to take special
+considerations.
+
+The tool `sdbootutil` is the one responsible of synchronizing the
+content in the rootfs with the ESP (kernel, bootloader, shim, boot
+entries, initrd, etc), so certain actions should be delegated to it.
+
+To complicate a bit more the situation, the transactional systems
+(like MicroOS) cannot now access the ESP from inside the transaction
+as they will break the atomicity update operation.  It is the
+`snapper` plugin provided by `sdbootutil` the one that will call the
+scripts in the correct moment (when setting the new default snapshot).
+
+This impose that `weak-modules2`, `regenerate-initrd-posttrans`, the
+`kernel-scriptlets` and the snapper plugin now need to work in
+coordination, reordering certain actions depending on the kind of
+system.  The next table will try to summarize those interaction:
+
+
```

Comment:
> The order of the actions is confusing in this table 

You are right. I changed the order.

> Also, the "Kernel/initrd" and "KMP/initrd" operations simply seem to just indicate the creation of the marker file under /run/regenerate-initrd?

Yes. But they are done in different scriptlets.

> I suppose that when you say "MicroOS" in the table below, you mean any system using transactional-update? 

This is correct. I used MicroOS because is shorter, but I will add a clarification note.

Should be done